explainaboard/analysis/analyses.py

```diff
@@ -321,6 +333,264 @@ def perform(
+            expected_calibration_error=expected_calibration_error,
+            maximum_calibration_error=maximum_calibration_error,
+        )
+
+    def __post_init__(self):
+        """Set the class name and validate."""
+        for bucket_perf in self.bucket_performances:
+            accuracy_performance = bucket_perf.performances.get("Accuracy", None)
+            if not accuracy_performance:
+                raise ValueError(
+                    "Wrong metrics. "
+                    "Required: Accuracy, "
+                    f"got: {set(bucket_perf.performances.keys())}"
+                )
+            if not isinstance(
+                accuracy_performance.auxiliary_result, ConfidenceMetricResult
+            ):
+                raise ValueError(
+                    "Wrong accuracy auxiliary result. "
+                    "Required: ConfidenceMetricResult, "
+                    f"got: {type(accuracy_performance.auxiliary_result)}"
+                )
+
+        self.cls_name: str = self.__class__.__name__
+
+    def generate_report(self) -> str:
+        """See AnalysisResult.generate_report."""
+        texts: list[str] = []
+
+        metric_names = sorted(k for k in self.bucket_performances[0].performances)
+
+        for metric_name in metric_names:
+            texts.append(f"the information of #{self.name}#")
+            texts.append(f"bucket_name\t{metric_name}\t#samples")
+
+            for bucket_perf in self.bucket_performances:
+                perf = bucket_perf.performances[metric_name]
+
+                if bucket_perf.bucket_interval is not None:
+                    bucket_name = f"{unwrap(bucket_perf.bucket_interval)}"
+                else:
+                    bucket_name = unwrap(bucket_perf.bucket_name)
+
+                texts.append(
+                    f"{bucket_name}\t" f"{perf.value}\t" f"{bucket_perf.n_samples}"
+                )
+
+            texts.append('')
+
+        texts.append(f"expected_calibration_error\t{self.expected_calibration_error}")
+        texts.append(f"maximum_calibration_error\t{self.maximum_calibration_error}")
+        texts.append('')
+        return "\n".join(texts)
+
+
+@final
+@dataclass
+class CalibrationAnalysis(Analysis):
+    """Perform calibration analysis.
+
+    The interval [0, 1] is evenly divided into buckets.
+    Calculate the accuracy and average confidence of each bucket.
+
+    Attributes:
+        feature: the name of the confidence feature
+        num_buckets: the number of buckets to be used
+        sample_limit: an upper limit on the number of samples saved in each bucket.
+        cls_name: the name of the class.
+    """
+
+    feature: str
+    num_buckets: int = 10
+    sample_limit: int = 50
+    cls_name: Optional[str] = None
+
+    def __post_init__(self):
+        """Set the class name."""
+        self.cls_name: str = self.__class__.__name__
+        if self.num_buckets <= 0:
+            self.num_buckets = 10
```

Comment:
Raise error rather than implicitly setting defaults. Implicit setting often causes unintended behavior.
```suggestion
        if self.num_buckets <= 0:
            raise ValueError(f"Invalid num_buckets: {num_buckets}")
```